### PR TITLE
feat: Add node events based metrics

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Adding metrics based on node events
- Metric: `operator_node_event_total`

**How was this change tested?**
- N/A

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: pkg/controllers/controllers.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"context"
 
 	"github.com/awslabs/operatorpkg/controller"
+	opevents "github.com/awslabs/operatorpkg/events"
```
Comment (rschalo) [inline pkg/controllers/controllers.go:21]: nit: just use events if there aren't collisions.

Comment (engedaam) [reply pkg/controllers/controllers.go:21]: We do have a collisions https://github.com/aws/karpenter-provider-aws/pull/7382/files#diff-b32dd54d2bd7ecc467b19203287df80c4c1e3934ea80602aadd6658306b5cca6L43

**Review by njtran — APPROVED** [review]

**Pull request merged**